smack-extensions/src/test/java/org/jivesoftware/smackx/dataformmedia/MediaElementTest.java

```diff
@@ -0,0 +1,106 @@
+/**
+ *
+ * Copyright 2019 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.dataformmedia;
+
+import static org.jivesoftware.smack.test.util.XmlUnitUtils.assertXmlSimilar;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+
+public class MediaElementTest {
+
+    @Test
+    public void toXmlTestForAudio() {
+        String xml = "<media xmlns='urn:xmpp:media-element'>" +
+                "<uri type='audio/x-wav'>" +
+                "http://victim.example.com/challenges/speech.wav?F3A6292C" +
+                "</uri>" +
+                "<uri type='audio/ogg; codecs=speex'>" +
+                "cid:sha1+a15a505e360702b79c75a5f67773072ed392f52a@bob.xmpp.org" +
+                "</uri>" +
+                "<uri type='audio/mpeg'>" +
+                "http://victim.example.com/challenges/speech.mp3?F3A6292C" +
+                "</uri>" +
+                "</media>";
+        URI uri1 = null;
+        URI uri2 = null;
+        URI uri3 = null;
+
+        try {
+            uri1 = new URI("http://victim.example.com/challenges/speech.wav?F3A6292C");
+            uri2 = new URI("cid:sha1+a15a505e360702b79c75a5f67773072ed392f52a@bob.xmpp.org");
+            uri3 = new URI("http://victim.example.com/challenges/speech.mp3?F3A6292C");
+        } catch (URISyntaxException e) {
+             throw new IllegalArgumentException("Invalid URI");
```

Comment:
I'd rather throw an `AssertionError` here, but that doesn't really matter too much.

Reply:
Besides from swallowing `e` here, why would you catch the `URISyntaxException` anyways? Just declare that the method throws it.